icaruscode/Filters/FilterCRTPMTMatching_module.cc

```diff
@@ -13,556 +13,607 @@
+            fGeometryService->OpDetGeoFromOpChannel(hit->OpChannel())
+                .GetCenter();
+        double amp = hit->Amplitude();
+        ampsum += amp;
+        fHitX.push_back(pos.X());
+        fHitY.push_back(pos.Y());
+        fHitZ.push_back(pos.Z());
+        fHitT.push_back(hit->PeakTime());
+        fHitA.push_back(amp);
+        flashCentroid.add(pos, amp);
+        t_m = t_m + hit->PeakTime();
+      }
+      geo::Point_t flash_pos = flashCentroid.middlePoint();
+      t_m = t_m / nPMTsTriggering;
+      if (nPMTsTriggering < fnOpHitToTrigger) {
+        continue;
+      }
+      bool inTime = flashInTime(firstTime, m_gate_type, m_trigger_gate_diff,
+                                m_gate_width);
+      double fThisRelGateTime = m_trigger_gate_diff + (tAbsflash - 1500) * 1e3;
+      bool fThisInTime_gate = false;
+      bool fThisInTime_beam = false;
+      if (m_gate_type == 1 || m_gate_type == 3) {  // BNB OffBeamBNB
+        if (fThisRelGateTime > fBNBBeamGateMin &&
+            fThisRelGateTime < fBNBBeamGateMax)
+          fThisInTime_gate = true;
+        if (fThisRelGateTime > fBNBinBeamMin &&
+            fThisRelGateTime < fBNBinBeamMax)
+          fThisInTime_beam = true;
+      }
+      if (m_gate_type == 2 || m_gate_type == 4) {  // NuMI OffBeamNuMI
+        if (fThisRelGateTime > fNuMIBeamGateMin &&
+            fThisRelGateTime < fNuMIBeamGateMax)
+          fThisInTime_gate = true;
+        if (fThisRelGateTime > fNuMIinBeamMin &&
+            fThisRelGateTime < fNuMIinBeamMax)
+          fThisInTime_beam = true;
+      }
+      inTime = fThisInTime_gate;
+      icarus::crt::CRTMatches CRTmatches = CRTHitmatched(
+          firstTime, flash_pos, crtHitList, fTimeOfFlightInterval);
+      int TopEn = 0, TopEx = 0, SideEn = 0, SideEx = 0;
+      auto nCRTHits = CRTmatches.entering.size() + CRTmatches.exiting.size();
+      std::vector<MatchedCRT> thisFlashCRTmatches;
+      eventType = CRTmatches.matchType;
+      if (nCRTHits > 0) {
+        for (auto& entering : CRTmatches.entering) {
+          vector<double> CRTpos = {entering.CRTHit->x_pos,
+                                   entering.CRTHit->y_pos,
+                                   entering.CRTHit->z_pos};
+          geo::Point_t thisCRTpos = {entering.CRTHit->x_pos,
+                                     entering.CRTHit->y_pos,
+                                     entering.CRTHit->z_pos};
+          double CRTtime = entering.CRTHit->ts1_ns / 1e3;
+          int CRTRegion = entering.CRTHit->plane;
+          int CRTSys = 0;
+          if (CRTRegion >= 36)
+            CRTSys =
+                1;  // Very lazy way to determine if the Hit is a Top or a Side.
+                    // Will update it when bottom CRT will be availble.
+          double CRTTof_opflash = CRTtime - tflash;
+          std::vector<int> HitFebs;
+          for (auto& crts : entering.CRTHit->feb_id) {
+            HitFebs.emplace_back((int)crts);
+          }
```

Comment:
Prefer copy for small data types:
```suggestion
          for (auto crts : entering.CRTHit->feb_id) {
            HitFebs.emplace_back((int)crts);
          }
```